test.js

```diff
@@ -74,6 +74,40 @@ test('promise support', async t => {
 	t.is(await memoized(10), 1);
 });
 
+test('do not cache rejected promises', async t => {
+	let i = 0;
+	const memoized = m(() => {
+		i++;
+		if (i === 1) {
+			return Promise.reject(new Error('foo bar'));
+		}
+
+		return Promise.resolve(i);
+	});
```

Comment:
This can be simplified by using an async function:

```js
const memoized = m(async () => {
	i++;

	if (i === 1) {
		throw new Error('foo bar');
	}

	return i;
}, {
	cacheRejection: true
});
```

Reply:
Much better :)